plover/machine/base.py

```diff
@@ -137,17 +137,29 @@ def get_option_info(cls):
         """Get the default options for this machine."""
         return {}
 
+class _ThreadedStenotypeThread(threading.Thread):
+    def __init__(self, machine):
+        threading.Thread.__init__(self)
+        self.machine = machine
+
+    def run(self):
+        try:
+            self.machine.run()
+        except Exception:
+            if not self.machine.finished.isSet():
+                log.error(self.machine.name + ' got disconnected.')
```

Comment:
Can we simplify the message to just "GeminiPr disconnected"?

```suggestion
                log.error('%s disconnected', self.machine.name)
```

Too bad there's no easy way to get the plugin name from what I can see.